### PR TITLE
refactor: move IERC20 lib to its own file & remove warnings

### PR DESCRIPTION
**Pull request by adhusson**

### CODE REVIEW

File: src/MgvLib.sol [hunk 1]
```diff
@@ -187,25 +188,3 @@ interface IMgvMonitor {
 
   function read(address outbound_tkn, address inbound_tkn) external view returns (uint gasprice, uint density);
 }
-
-interface IERC20 {
```
Comment (lnist) [inline src/MgvLib.sol:191]: So this is changing audited code, but it is just moving stuff around. Is that ok?

File: src/IERC20.sol [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  function transferFrom(address sender, address recipient, uint amount) external returns (bool);
+
+  function symbol() external view returns (string memory);
+
+  event Transfer(address indexed from, address indexed to, uint value);
+  event Approval(address indexed owner, address indexed spender, uint value);
+
+  function decimals() external view returns (uint8);
+
+  function name() external view returns (string memory);
```
Comment (lnist) [inline src/IERC20.sol:24]: Its just moving stuff around, and adding this method? :)

**Review by lnist — APPROVED** [review]

File: src/MgvLib.sol (hunk 1 above)
Comment (adhusson) [reply src/MgvLib.sol:191]: Damn I misremembered that MgvLib was not audited. Yes I guess :/

File: src/IERC20.sol (hunk 2 above)
Comment (adhusson) [reply src/IERC20.sol:24]: Yes I noticed that it was missing, even though it is part of the [ERC20](https://eips.ethereum.org/EIPS/eip-20) standard

**Pull request merged**